### PR TITLE
[201811][platform/cel]: Port fancontrol changes for dx010/e1031 to 201811 branch

### PR DESCRIPTION
**Pull request by mudsut4ke**
**- What I did**
-  Update fancontrol service for Seastone-DX010/E1031 device to support hysteresis temperature threshold and difference config for each unit fan direction type (B2F/F2B) follow master branch

**- How I did it**
- Add fancontrol config file fancontrol-B2F,fancontrol-F2B
- Add fancontrol script fancontrol.sh
- Modify fancontrol daemon to support i2c base config

**- How to verify it**
- Verify fancontrol service status
  - [dx010-201811-fsc-test.log](https://github.com/Azure/sonic-buildimage/files/4856362/dx010-201811-fsc-test.log)
  - [e1031-201811-fsc-test.log](https://github.com/Azure/sonic-buildimage/files/4856368/e1031-201811-fsc-test.log)

- Verify fan speed under thermal chamber 

Signed-off-by: Wirut Getbamrung wgetbumr@celestica.com

### CODE REVIEW

**Comment by mudsut4ke** [comment]
retest broadcom please

**Comment by mudsut4ke** [comment]
retest broadcom please

**Comment by mudsut4ke** [comment]
> There is inconsistent alignment in a number of these files due to a combination of tabs and spaces. Please replace all tabs with four spaces throughout.

@jleveque , updated in https://github.com/Azure/sonic-buildimage/pull/4867/commits/a5d529dfbf2fc4b0d2808c53f57a10f07afc5996

File: platform/broadcom/sonic-platform-modules-cel/haliburton/scripts/fancontrol.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+DEVPATH=/sys/devices/pci0000:00/0000:00:13.0/i2c-0/i2c-8/i2c-23/23-004d 
+
+test -x $DAEMON || exit 0
+
+for i in 1 2 3
+do
+    j=$i
+    [ $i -eq 3 ] && j=4
+    FANFAULT=$(cat ${DEVPATH}/fan${j}_fault)
+  [ $FANFAULT = 1 ] && continue
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-cel/haliburton/scripts/fancontrol.sh:32]: Incorrect indentation

File: platform/broadcom/sonic-platform-modules-cel/haliburton/scripts/fancontrol.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    FANDIR=$(cat ${PLATFORMPATH}/fan${i}_dir)
+done
+CONF=${MAIN_CONF}-${FANDIR}
+
+case "$1" in
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-cel/haliburton/scripts/fancontrol.sh:37]: Indentation inside this case statement is 2 spaces; should be 4.

**Pull request merged**